events/external_handler.go

```diff
@@ -0,0 +1,114 @@
+package events
```

Comment:
This file, along with util.go and worker.go are not really new. I just pulled them out of listener.go into their own files because that file was becoming too large and confusing.

I should have separated it into two commits, but was in a rush.

The interesting change is the contents of websocket_pings.go and the corresponding test.